server/controllers/locationController.ts

```diff
@@ -0,0 +1,95 @@
+import { Request, Response } from 'express';
+import { Like } from 'typeorm';
+
+import { Location } from 'server/models/Location';
+
+export default {
+  async index(req: Request, res: Response) {
+    const { country_code, city, region, postal_code, address } = req.query;
+
+    const locations = await Location.find({
+      where: [
+        { country_code: Like(`%${country_code}%`) },
+        { city: Like(`%${city}%`) },
+        { region: Like(`%${region}%`) },
+        { postal_code: Like(`%${postal_code}%`) },
+        { address: Like(`%${address}%`) },
+      ],
+    });
+
+    res.json(locations);
+  },
+
+  async show(req: Request, res: Response) {
+    const { id } = req.params;
+
+    const location = await Location.findOne({ id: parseInt(id) });
+
+    if (location) {
+      res.json(location);
+    } else {
+      res.status(404).json({ error: "Can't find location" });
+    }
+  },
+
+  async create(req: Request, res: Response) {
```

Comment:
This should be a protected route (only users can create)

Reply:
We'll add this later when we create the auth strategy (cookies/jwt/etc...)

Reply:
Doesn't passport handle it all the same way anyway regardless of strategy? By adding `req.user` to the request?

Reply:
Yes it does, but I don't think we should add passport in this PR, maybe better in a separate PR. Also we need to do the same for other API routes, so better to do it in a different PR

Reply:
I agree that we shouldn't add passport in this PR, but it would still be good to implement.  I'll address changes in #344 and once that's merged we can implement it here and test it accordingly.  No point in omitting it since we're not building the frontend for it anyway.

In any case, we should probably prioritize getting passport in since so many things depend on it.

Reply:
Yeah we are, I'm doing it parallel in #342, So I'd like to get this merge so I can work on that, #341 depends on this and #342 and I'd like to get this done before tomorrow's meeting

Reply:
Ah didn't realize you were building it.  Fine by me to defer this then!